src/main/kotlin/me/serce/solidity/lang/resolve/engine.kt

```diff
@@ -180,6 +180,14 @@ object SolResolver {
     return result + result.map { collectImports(it.file.childrenOfType<SolImportDirective>(), visited) }.flatten()
   }
 
+  //collect all SolContractDefinition recursively from imports
+  fun collectContracts(file: PsiFile): Collection<SolContractDefinition> {
```

Comment:
please merge the master into your branch.
there was `collectImports` method added within https://github.com/intellij-solidity/intellij-solidity/pull/450
it looks like that method do the same thing, but it also contains a recursive call guard

Reply:
Indeed it's much better, thanks !
I already merged main into my branch